src/bot.jl

```diff
@@ -8,24 +8,36 @@ const UStrings = Union{AbstractString,Regex,AbstractChar}
 Config object that holds the options and configuration for the SnoopCompile bot. This object is fed to the `@snoopi_bot`.
 
 # Arguments:
-- `packageName::String`
-- `subst::Vector{Pair{UStrings, UStrings}}` : to replace a packages precompile setences with another's package like `["ImageTest" => "Images"]`
-- `blacklist::Vector{UStrings}` : to remove some precompile sentences
+- `package_name::String`
 
-`const UStrings ==  Union{AbstractString,Regex,AbstractChar}` # every string like type that `replace()` has a method for.
+## optional:
+
+- `subst` : A vector of pairs of Strings (or RegExp) to replace a packages precompile setences with another's package like `["ImageTest" => "Images"]`.
+
+- `blacklist` : A vector of of Strings (or RegExp) to remove some precompile sentences
+
+- `os`: A vector of of Strings (or RegExp) to give the list of os that you want to generate precompile signatures for. Each element will call a `Sys.is\$eachos()` function.
+
+
+# Example
+```julia
+BotConfig("MatLang", blacklist = ["badfunction"], os = ["linux", "windows"])
+```
 """
 struct BotConfig
-    packageName::String
+    package_name::String
     subst::Vector{Pair{T1, T2}} where {T1<:UStrings, T2 <: UStrings}
     blacklist::Vector{T3} where {T3<:UStrings}
+    os::Union{Vector{String}, Nothing}
 end
 
-function BotConfig(packageName::String; subst::Vector{Pair{T1, T2}} where {T1<:UStrings, T2 <: UStrings} = Vector{Pair{String, String}}(), blacklist::Vector{T3} where {T3<:UStrings}= String[])
-    return BotConfig(packageName, subst, blacklist)
+function BotConfig(package_name::String; subst::Vector{Pair{T1, T2}} where {T1<:UStrings, T2 <: UStrings} = Vector{Pair{String, String}}(), blacklist::Vector{T3} where {T3<:UStrings}= String[], os::Union{Vector{String}, Nothing} = nothing)
```

Comment:
```suggestion
function BotConfig(package_name::AbstractString; subst::AbstractVector = Vector{Pair{UString, UString}}(), blacklist::AbstractVector= UString[], os::Union{Vector{String}, Nothing} = nothing)
```
If the user passes in some other kind of compatible object, Julia will convert it for you.